napari/utils/_tests/test_progress.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+import pytest
+
+try:
+    import qtpy  # noqa
+except ImportError:
+    pytest.skip('Cannot test progress without qtpy.', allow_module_level=True)
+except RuntimeError:
+    pytest.skip(
+        'Cannot test progress without Qt bindings.', allow_module_level=True
+    )
+from ..progress import ProgressBar, progress
+
+
+def activity_dock_children(viewer):
+    return viewer.window.qt_viewer.activityDock.children()[4].children()
+
+
+def test_progress_with_iterable(make_napari_viewer):
+    viewer = make_napari_viewer()
+    r = range(100)
+    pbr = progress(r)
+
+    assert any(
+        isinstance(wdg, ProgressBar) for wdg in activity_dock_children(viewer)
+    )
+    assert pbr.iterable is r
+    assert pbr.n == 0
+    assert pbr._pbar.pbar.maximum() == pbr.total == 100
+
+
+def test_progress_with_ndarray(make_napari_viewer):
```

Comment:
The reason you're getting the `RuntimeError: wrapped C/C++ object of type ProgressBar has been deleted` error is because tqdm calls `self.close()` at object deletion right [here](https://github.com/tqdm/tqdm/blob/f86104a1f30c38e6f80bfd8fb16d5fcde1e7749f/tqdm/std.py#L1144-L1145).  However, all QWidgets close and delete their child widgets when they are closed.  So what's happening here is that the viewer is getting closed at the end of the test... which calls `pbar.close()` ... and then at some later point, the `progress` python object is deleted, calling `del progress`, which tries to close `progress._pbar` _again_ and you get the error.

This can be solved multiple ways, but basically it all comes down to making sure that you don't call `ProgressBar.close()` twice.  Adding `pbr.close()` to the end of the test would solve it.  Wrapping `self._pbar.close()` in a `try/except RuntimeError` would solve it (but be a bit messier).

But most importantly, this is one of the main reasons why I mentioned in [this comment](https://github.com/DragaDoncila/napari/pull/1#pullrequestreview-637133744) that we should try avoid using `pbar = progress()` directly (preferring instead to use either `for i in progress()` or using `with progress() as pbar:`)   ... and wherever we _do_ use it, always accompany it with `pbar.close()`

Reply:
Ok this makes sense to me now. I couldn't understand how the `ProgressBar` could have been deleted without going through `progress.close` but of course the viewer has its own cleanup.